airflow/cli/commands/kubernetes_command.py

```diff
@@ -90,7 +90,25 @@ def cleanup_pods(args):
     print('Loading Kubernetes configuration')
     kube_client = get_kube_client()
     print(f'Listing pods in namespace {namespace}')
-    list_kwargs = {"namespace": namespace, "limit": 500}
+    airflow_pod_labels = [
+        'airflow-worker',
+        'dag_id',
+        'task_id',
+        'execution_date',
+        'try_number',
+        'airflow_version',
+        'kubernetes_executor',
+    ]
+    list_kwargs = {
+        "namespace": namespace,
+        "limit": 500,
+        "label_selector": client.V1LabelSelector(
+            match_expressions=[
+                client.V1LabelSelectorRequirement(key=label, operator="Exists")
```

Comment:
> The requirements are ANDed.

This command should also work when the pod does not contain `kubernetes_executor` label i.e. it is started by KubernetetsPodOperator.

Reply:
Makes sense to me. For the same reason, if I'm not wrong, we should exclude considering `airflow-worker` as well.

Addressed in https://github.com/apache/airflow/pull/15204/commits/e406a0895a173587fd0cff8354650c740e687fd1

Thanks!